projects/vendor/src/yotpo/baseyotpo/baseyotpo.service.ts

```diff
@@ -0,0 +1,37 @@
+import { Injectable, ElementRef } from '@angular/core';
+import { Observable } from 'rxjs';
+import { Product, ProductService, RoutingService, WindowRef } from '@spartacus/core';
+import { filter, map, switchMap } from 'rxjs/operators';
+import { YotpoConfig } from '../yotpoconfig/yotpo-config';
+
+@Injectable({
+ providedIn: 'root',
+})
+export abstract class BaseyotpoService  {
+  constructor(protected config: YotpoConfig, protected windowRef: WindowRef,
+    protected routingService: RoutingService,
+    protected productService: ProductService
+  ) {
+    const widgetscript = this.windowRef.document.createElement('script');
+    widgetscript.type = 'text/javascript';
+    widgetscript.async = true;
+    widgetscript.src = `https://staticw2.yotpo.com/${this.config.vendor.yotpo.appToken}/widget.js`;
+    this.windowRef.document.getElementsByTagName('head')[0].appendChild(widgetscript);
+}
+
+  getProduct(): Observable<Product> {
+    return this.routingService.getRouterState().pipe(
+      map(state => state.state.params['productCode']),
+      filter(productCode => !!productCode),
+      switchMap((productCode: string) => this.productService.get(productCode))
+    );
+  }
+
+  afterViewInit(elementRef: ElementRef) {
```

Comment:
Could you give this function a name that provides a better description of the usage.

Reply:
done